src/lp_pricer/lp_pricer.cairo

```diff
@@ -0,0 +1,391 @@
+            let old_admin = Ownable::OwnableImpl::owner(@state);
+            // [Check] New admin is not already registered
+            assert(new_admin_address != old_admin, errors::ALREADY_ADMIN);
+            // [Check] New admin is not zero
+            assert(!new_admin_address.is_zero(), errors::ZERO_ADDRESS_ADMIN);
+
+            // [Effect] Transfer ownership
+            Ownable::OwnableImpl::transfer_ownership(ref state, new_admin_address);
+
+            // [Interaction] Transfered ownership event
+            self
+                .emit(
+                    Event::TransferOwnership(
+                        TransferOwnership { old_address: old_admin, new_address: new_admin_address }
+                    )
+                );
+        }
+
+        /// Returns the admin address.
+        fn get_admin_address(self: @ContractState) -> ContractAddress {
+            let state: Ownable::ContractState = Ownable::unsafe_new_contract_state();
+            let res = Ownable::OwnableImpl::owner(@state);
+            res
+        }
+
+        /// Returns the Pragma Oracle address.
+        fn get_oracle_address(self: @ContractState) -> ContractAddress {
+            self.get_oracle_address()
+        }
+    }
+
+    // ================== PRIVATE FUNCTIONS ==================
+    // Utilities used in the Public ABI.
+
+    /// Asserts that the caller is the Admin.
+    fn assert_only_admin() {
+        let state: Ownable::ContractState = Ownable::unsafe_new_contract_state();
+        let admin = Ownable::OwnableImpl::owner(@state);
+        let caller = get_caller_address();
+        assert(caller == admin, errors::NOT_ADMIN);
+    }
+
+    /// Retrieves the token addresses from from the Pool Dispatcher.
+    fn fetch_tokens_addresses(
+        pool_dispatcher: IPoolDispatcher
+    ) -> (ContractAddress, ContractAddress) {
+        (pool_dispatcher.token_0(), pool_dispatcher.token_1())
+    }
+
+    /// Retrieves the token symbols from the underlying currencies.
+    fn fetch_tokens_symbols(
+        token_a_address: ContractAddress, token_b_address: ContractAddress
+    ) -> (felt252, felt252) {
+        let token_a = ERC20CamelABIDispatcher { contract_address: token_a_address };
+        let token_b = ERC20CamelABIDispatcher { contract_address: token_b_address };
+        (token_a.symbol(), token_b.symbol())
+    }
+
+    /// Returns true if the currency is supported by Pragma, else false.
+    fn currency_is_supported(oracle: IOracleABIDispatcher, currency_id: felt252) -> bool {
+        oracle.get_currency(currency_id).id != 0
+    }
+
+    /// Returns the price in USD for a currency by fetching it from the Pragma Oracle.
+    fn get_currency_price_in_usd(oracle: IOracleABIDispatcher, currency_id: felt252) -> u256 {
+        let pair_id = StringTrait::concat(currency_id, USD_PAIR_SUFFIX);
+        let data_type = DataType::SpotEntry(pair_id);
+        let data = oracle.get_data_median(data_type);
+        data.price.into()
```

Comment:
it can be interesting to add a check in case the price returned is null (unlikely to happen, but for better safety)